### PR TITLE
Record audio using MP3 instead of WEBM

### PR DESCRIPTION
**Pull request by ansonmiu0214**
Addresses cockatoos/sprint-board#25

This PR adapts the MediaRecorder API to record using MP3 encoding (rather than .webm) using a WebAssembly library,
and sends the audio blob to an Azure function to be ∂ecoded into an MP3 file for the accent scoring stage.

### CODE REVIEW

File: src/app/components/article-comparison/article-comparison.component.ts [hunk 1]
```diff
@@ -38,24 +43,40 @@ export class ArticleComparisonComponent implements OnInit, OnChanges {
...
+        // this.convertApiUrl = "http://127.0.0.1:5000/blob2mp3";
+        this.convertApiUrl = "https://conversiontest.azurewebsites.net/api/httptrigger1";
+        this.recordingEncoding$.subscribe((base64Encoding) => {
+            if (base64Encoding === undefined) {
+                return;
+            }
+
+            // TODO: send base64 encoding to Azure :)
+            this.http.post(this.convertApiUrl, { blob: base64Encoding }, { responseType: "text" }).subscribe((res) => {
+                console.log(res);
```
Comment (ansonmiu0214) [inline src/app/components/article-comparison/article-comparison.component.ts:67]: I think we should notify the user (via the UI) when the audio recording has been processed.
But we can leave this feature for another UI story.

File: src/app/components/article-comparison/article-comparison.component.ts [hunk 2]
```diff
@@ -38,24 +43,40 @@ export class ArticleComparisonComponent implements OnInit, OnChanges {
     // Flag to signal if the text-to-speech service is speaking.
     isSpeaking$: Observable<boolean>;
 
-    constructor(private store: Store<AppState>) {
+    // Base64 encoding of the user's recording.
+    recordingEncoding$: Observable<string>;
+
+    // Web server/Azure function endpoint for converting recording.
+    // TODO: Any better place for this to live?
+    convertApiUrl: any;
```
Comment (ansonmiu0214) [inline src/app/components/article-comparison/article-comparison.component.ts:52]: Put this under `environments/environment.ts` (development) and `environments/environment.prod.ts` (production). 

Then run `ng serve` to use the local environment stuff (e.g. localhost), and run `ng serve -c production` to use the production environment API

File: src/app/components/practice-container/practice-container.component.ts [hunk 3]
```diff
@@ -14,7 +15,8 @@ export class PracticeContainerComponent implements OnInit {
     article$: Observable<Article>;
 
     constructor(public dialog: MatDialog, public articlesService: ArticlesService) {
-        this.article$ = articlesService.getDocument();
+        // this.article$ = articlesService.getDocument();
```
Comment (ansonmiu0214) [inline src/app/components/practice-container/practice-container.component.ts]: If you un-comment Line 18 and comment out Line 19, does it work for you?

**Review by ansonmiu0214 — COMMENTED** [review]
@jyz16 a couple of changes, but otherwise LGTM :)

File: src/app/components/article-comparison/article-comparison.component.ts (hunk 1 above)
Comment (jyz16) [reply src/app/components/article-comparison/article-comparison.component.ts:67]: Ah good idea! Taking note to include this in Azure Function #14

File: src/app/components/practice-container/practice-container.component.ts (hunk 3 above)
Comment (jyz16) [reply src/app/components/practice-container/practice-container.component.ts]: Yep I think so, do you want me to change it?

Comment (ansonmiu0214) [reply src/app/components/practice-container/practice-container.component.ts]: Yes please :)

**Comment by ansonmiu0214** [comment]
Thanks, LGTM! Will merge, :)

**Pull request merged**